internal/storage/image.go

```diff
@@ -407,9 +409,32 @@ type copyImageArgs struct {
 	StoreOptions storage.StoreOptions
 }
 
+// maybeMoveSelfToCgroup moves the current process to transient cgroup when running under systemd.
+func maybeMoveSelfToCgroup() error {
+	// If CRI-O is not already running under systemd, do not attempt
+	// to setup a cgroup.
+	if os.Getenv("INVOCATION_ID") == "" {
+		return nil
+	}
+
+	slice := "system.slice"
+	if rootless.IsRootless() {
+		slice = "user.slice"
+	}
+
+	unitName := fmt.Sprintf("crio-pull-image-%d.scope", os.Getpid())
```

Comment:
Why the new unique unit name for every copyImage? I mean, can we just use `crio-pull-image.scope` for all of them?

Reply:
The goal is to run each pull in a separate cgroup, so we could apply limits if needed in future.

I've not looked if it is possible, but can we join an existing systemd scope?

Reply:
> I've not looked if it is possible, but can we join an existing systemd scope?

Not sure about systemd, but we can join the cgroup. That will be a kludge though :(

In general I think it make more sense for all pulls to be run within one single scope. That might require a different architecture though :-\

Reply:
Eventually, we want to charge image pulls to a pod cgroup once k8s upstream has the plumbing to give us a pod cgroup for an image pull.